File: back-end/app/services/imageService.js
import Image from "../models/Image.js";

class ImageService {
  constructor() {
    this.model = Image;
  }

  async getAllImages() {
    return await this.model.find();
  }

  async getImageById(id) {
    return await this.model.findById(id);
  }

  async createImage(imageData) {
    const image = new this.model(imageData);
    return await image.save();
  }

  async updateImage(id, updateData) {
    return await this.model.findByIdAndUpdate(id, updateData, { new: true });
  }

  async deleteImage(id) {
    return await this.model.findByIdAndDelete(id);
  }
}

export default new ImageService();

File: server/src/main/java/com/instagram/server/service/ImageService.java
package com.instagram.server.service;

import com.instagram.server.entity.Image;
import com.instagram.server.repository.ImageRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.time.LocalDateTime;
import java.util.List;
import java.util.Optional;

@Service
public class ImageService {

    @Autowired
    private ImageRepository imageRepository;

    public List<Image> getAllImages() {
        return imageRepository.findAll();
    }

    public Optional<Image> getImageById(String id) {
        return imageRepository.findById(id);
    }

    public Image createImage(Image image) {
        return imageRepository.save(image);
    }

    public Image updateImage(String id, Image imageDetails) {
        Optional<Image> optionalImage = imageRepository.findById(id);
        if (optionalImage.isPresent()) {
            Image image = optionalImage.get();
            image.setImage(imageDetails.getImage());
            image.setUpdatedAt(LocalDateTime.now());
            return imageRepository.save(image);
        } else {
            throw new RuntimeException("Image not found with id " + id);
        }
    }

    public void deleteImage(String id) {
        imageRepository.deleteById(id);
    }
}
